resources/models.py

```diff
@@ -663,21 +665,37 @@ def get_queryset(self):
         # Pre-filter the leases for the upcoming/current season
         renewed_leases = BerthLease.objects.filter(
             in_current_season, berth=OuterRef("berth"), customer=OuterRef("customer"),
-        )
+        ).values("pk")
         # Filter the leases from the previous season that have already been renewed
-        previous_leases = BerthLease.objects.exclude(Exists(renewed_leases)).filter(
-            in_last_season, paid_status, berth=OuterRef("pk")
+        previous_leases = (
+            BerthLease.objects.exclude(Exists(renewed_leases))
+            .filter(in_last_season, paid_status, berth=OuterRef("pk"))
+            .values("pk")
         )
 
         # For the leases that have been renewed or are valid during the current season.
         # Filter the leases on the current season that have not been rejected
         current_leases = BerthLease.objects.filter(
             in_current_season, active_current_status, berth=OuterRef("pk")
-        )
+        ).values("pk")
 
         # A berth is NOT available when it already has a lease on the current (or upcoming) season
-        # or when the previous season lease has been paid and the new leases have not been sent
-        is_available = ~Exists(previous_leases | current_leases)
+        # or when the previous season lease has been paid and the new leases have not been sent.
+        active_leases = ~Exists(previous_leases | current_leases)
+
+        # Additionally, the berth is also NOT available when there is a switch offer drafted or offered
+        # (this requires separate Exists clauses
+        active_offers = ~Exists(
+            BerthSwitchOffer.objects.filter(
+                status__in=(OfferStatus.DRAFTED, OfferStatus.OFFERED),
+                berth=OuterRef("pk"),
+            ).values("pk")
+        )
+
+        # Need to explicitly mark the result of the AND as a BooleanField
+        is_available = ExpressionWrapper(
```

Comment:
Haven't used ExpressionWrapper myself, nice to learn something new :-)

Reply:
I don't think I had either, definitely a good way to learn on the go 🏃